### PR TITLE
Fix relation problem in hand-built uncommitted bundles

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by makyo** [comment]
# QA

QA can be done in devel/sandbox.

Search for and deploy the apache-hadoop bundle, ensure that relation lines show up after committing the bundle. Ensure that other bundles, such as the spark-cluster bundle deploy as well.

**Comment by jaycee** [comment]
Code looks :+1:  No QA. 

Sent from a mobile device with tiny keys. Forgive typos. 




On Fri, Jul 31, 2015 at 2:08 PM -0700, "Madison Scott-Clary" <notifications@github.com> wrote:










QA



QA can be done in devel/sandbox.



Search for and deploy the apache-hadoop bundle, ensure that relation lines show up after committing the bundle. Ensure that other bundles, such as the spark-cluster bundle deploy as well.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2663/
Test PASSed.

File: app/store/env/go.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ YUI.add('juju-env-go', function(Y) {
   var utils = Y.namespace('juju.views.utils');
 
   var endpointToName = function(endpoint) {
+    if (!endpoint[1].name) {
```
Comment (hatched) [inline app/store/env/go.js]: Can you add a detailed comment here explaining why this is necessary

File: test/test_env_go.js [hunk 2]
```diff
@@ -30,6 +30,19 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+    it('provides a way to retrieve a name from an endpoint', function() {
+      var endpointA = [
+        'foo',
+        { name: 'bar' }
+      ];
+      var endpointB = [
+        'bar',
+        {}
+      ];
+      assert.equal('foo:bar', environments.endpointToName(endpointA));
```
Comment (hatched) [inline test/test_env_go.js]: could you also add some comments here as to what this needs to be testing for.

**Comment by hatched** [comment]
:+1:  With some detailed comments as to why that code is necessary.

QA OK

I followed the steps in the QA instructions and it worked as expected. I also tried a few other bundles and it all appears to still work as expected.

**Comment by makyo** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2669/
Test PASSed.

**Pull request merged**